catalyst/dl/callbacks/metrics/ppv_tpr_f1.py

```diff
@@ -0,0 +1,92 @@
+from typing import List
+from collections import defaultdict
+
+import numpy as np
+import torch
+
+from catalyst.dl.meters import PrecisionRecallF1ScoreMeter
+from catalyst.dl.core import Callback, RunnerState, CallbackOrder
+
+
+class PrecisionRecallF1ScoreCallback(Callback):
+    """
+    Calculates the global precision (positive predictive value or ppv),
+    recall (true positive rate or tpr), and F1-score per class for each loader.
+    Currently, supports binary and multi-label cases.
+    """
+    def __init__(
+        self,
+        input_key: str = "targets",
+        output_key: str = "logits",
+        class_names: List[str] = None,
+        num_classes: int = 1,
+        threshold: float = 0.5
+    ):
+        """
+        Args:
+            input_key (str): input key to use for iou calculation
+                specifies our ``y_true``.
+            output_key (str): output key to use for iou calculation;
+                specifies our ``y_pred``
+            class_names (List[str]): class names to display in the logs.
+                If None, defaults to indices for each class, starting from 0.
+            num_classes (int): Number of classes
+            threshold (float): threshold for outputs binarization
+        """
+        super().__init__(CallbackOrder.Metric)
+        self.input_key = input_key
+        self.output_key = output_key
+
+        self.list_args = ["ppv", "tpr", "f1-score"]
+        self.class_names = class_names
+        self.num_classes = num_classes \
+            if class_names is None \
+            else len(class_names)
+
+        assert self.num_classes is not None
+
+        self.meters = [PrecisionRecallF1ScoreMeter(threshold)
+                       for _ in range(self.num_classes)]
+
+    def _reset_stats(self):
+        for meter in self.meters:
+            meter.reset()
+
+    def on_loader_start(self, state):
+        self._reset_stats()
+
+    def on_batch_end(self, state: RunnerState):
+        logits: torch.Tensor = state.output[self.output_key].detach().float()
+        targets: torch.Tensor = state.input[self.input_key].detach().float()
+        probabilities: torch.Tensor = torch.sigmoid(logits)
```

Comment:
I think, we could parametrise the used function during callback initialization

Reply:
I might be misinterpreting what you're saying, but are you asking to make a reusable function to do the ops above and initialize it as an attribute during callback initialization?

Reply:
I mean, you can parametrize activation function like [here](https://github.com/catalyst-team/catalyst/blob/master/catalyst/dl/callbacks/metrics/iou.py#L17)

Reply:
I think it would be more consistent, but I don't think `PrecisionRecallF1ScoreCallback` should inherit `MetricCallback` because `PrecisionRecallF1ScoreCallback` calculates the metrics for the entire loader instead of for each batch (maybe make a separate base global callback?). I'm curious on your thoughts?
(I'll add the option to specify the activation function but without the `super.init(...)` in the meantime)

Reply:
I think we can create new abstraction like "DatasetMetricCallback", that need need to collect statistic during `on_batch_end` and also some additional calculations `on_loader_end`. We have the same case with AUCCallback, so... looks like we need something general :)

Reply:
#450 <- Related PR

Reply:
I'll try to get it tomorrow!

Reply:
Do we need to add `.cpu()`? Otherwise it throws error:
`TypeError: can't convert CUDA tensor to numpy. Use Tensor.cpu() to copy the tensor to host memory first.`